### ISSUE TITLE
Cannot specify dependency on RELEASED version

### ISSUE THREAD

**Issue by packocz** [flxbl-io/sfp#18]
**Describe the bug**
When sfdx-project.json package definition specifies a dependency on another package's RELEASED version, the orchestrator fails with following error:
> Validating Org Authentication Mechanism..
>     Error: Unable to execute command .. MALFORMED_QUERY:
>     PatchVersion=1 and BuildNumber=RELEASED and ValidationSkipped

Example sfdx-project.json snippet:
```
"path": "src/case-management",
"default": false,
"package": "case-management",
"versionName": "ver 0.1",
"versionNumber": "0.1.0.NEXT",
"definitionFile": "config/package-def.json",
"dependencies": [
	{
		"package": "core",
		"versionNumber": "0.5.1.RELEASED"
	}
]
```
SF Documentation about dependency config allowing use of RELEASED
https://developer.salesforce.com/docs/atlas.en-us.pkg2_dev.meta/pkg2_dev/sfdx_dev_dev2gp_config_keywords.htm

**To Reproduce**
1. In a project with 2 packages with 1 depending on the other, set the dependency on major.minor.patch.RELEASED version
2. Promote the parent package at version from step 1
3. Run orchestrator script (tested with prepare pool)

**Expected behavior**
No errors

**Platform Details (please complete the following information):**
 - OS: Mac OS
 - Version: @dxatscale/sfpowerscripts/25.0.6 darwin-arm64 node-v20.8.0
 - Salesforce CLI(sfdx cli) Version: @salesforce/cli/2.28.6 darwin-arm64 node-v20.8.0
 - CI Platform: n\a

**Comment by Rocko1204** [comment]
Hi @packocz Fix/feature is in progress #20

**Comment by azlam-abdulsalam** [comment]
Done
